README.md

```diff
@@ -54,36 +72,47 @@ if __name__ == "__main__":
     main(MyCharm)
 ```
 
-This charm does nothing, though, so you'll typically want to observe some Juju
-events, such as `start`:
+This charm does nothing, because the `MyCharm` class passed to the operator
+framework's `main` function is empty. Functionality can be added to the charm
+by instructing it to observe particular Juju events when the `MyCharm` object
+is initialized. In the example below we're observing the `start` event; the
+known event sources are all attributes of `<emitter>.on`, so in this case
+`self.on.start` is passed to `self.framework.observe`. This will look for a
+method named `on_start` when that Juju event is triggered.
 
 ```python
 class MyCharm(CharmBase):
     def __init__(self, *args):
         super().__init__(*args)
-        self.framework.observe(self.on.start, self.on_start)
+        self.framework.observe(self.on.start, self)
```

Comment:
I believe Gustavo's feedback was that we should start with the more general case, where you do specify the name of the receiver function, and then maybe have a small note of an alternate syntax, but not made a prominent part of the Readme.